server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/EventDeadLettersIntegrationTest.java

```diff
@@ -0,0 +1,464 @@
+    private static final String SERIALIZED_GROUP = new RetryEventsListenerGroup().asString();
+
+    RetryEventsListener retryEventsListener = new RetryEventsListener();
+
+    @ClassRule
+    public static DockerCassandraRule cassandra = new DockerCassandraRule();
+
+    @Rule
+    public CassandraJmapTestRule cassandraJmapTestRule = CassandraJmapTestRule.defaultTestRule();
+
+    private GuiceJamesServer guiceJamesServer;
+    private MailboxProbe mailboxProbe;
+
+    private EventSerializer eventSerializer = new EventSerializer(new InMemoryId.Factory(), new InMemoryMessageId.Factory());
+
+    @Before
+    public void setUp() throws Exception {
+        retryEventsListener.clear();
+        guiceJamesServer = cassandraJmapTestRule.jmapServer(cassandra.getModule())
+            .overrideWith(new WebAdminConfigurationModule())
+            .overrideWith(binder -> Multibinder.newSetBinder(binder, MailboxListener.GroupMailboxListener.class).addBinding().toInstance(retryEventsListener));
+        guiceJamesServer.start();
+
+        DataProbe dataProbe = guiceJamesServer.getProbe(DataProbeImpl.class);
+        mailboxProbe = guiceJamesServer.getProbe(MailboxProbeImpl.class);
+        WebAdminGuiceProbe webAdminGuiceProbe = guiceJamesServer.getProbe(WebAdminGuiceProbe.class);
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminGuiceProbe.getWebAdminPort())
+            .build();
+
+        dataProbe.addDomain(DOMAIN);
+        dataProbe.addUser(BOB, BOB_PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        guiceJamesServer.stop();
+    }
+
+    @Test
+    public void failedEventShouldBeStoredInRetryListenerFailedEventsList() {
+        retryEventsListener.setRetriesBeforeSuccess(4);
+        mailboxProbe.createMailbox("#private", BOB, DefaultMailboxes.INBOX);
+
+        assertThat(retryEventsListener.getFailedEvents()).hasSize(1);
+    }
+
+    @Test
+    public void successfulEventShouldNotBeStoredInRetryListenerFailedEventsList() {
+        retryEventsListener.setRetriesBeforeSuccess(3);
+        mailboxProbe.createMailbox("#private", BOB, DefaultMailboxes.INBOX);
+
+        assertThat(retryEventsListener.getFailedEvents()).hasSize(0);
+    }
+
+    @Test
+    public void failedEventGroupShouldBeStoredInDeadLetter() {
+        retryEventsListener.setRetriesBeforeSuccess(4);
+        mailboxProbe.createMailbox("#private", BOB, DefaultMailboxes.INBOX);
+
+        when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups")
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .contentType(ContentType.JSON)
+            .body(".", containsInAnyOrder(SERIALIZED_GROUP));
+    }
+
+    @Test
+    public void failedEventShouldBeStoredInDeadLetter() {
+        retryEventsListener.setRetriesBeforeSuccess(4);
+        mailboxProbe.createMailbox("#private", BOB, DefaultMailboxes.INBOX);
+
+        Event failedEvent = retryEventsListener.getFailedEvents().get(0);
```

Comment:
Let's not rely on hacks to get the evnetId

After all we know that an event had been generated so we should have an event in deadletter, whatever its Id is. (Use `body(eventId, isNotNull())` matching.